services/ui-src/src/components/banners/AdminBannerProvider.tsx

```diff
@@ -37,12 +38,38 @@ export const AdminBannerProvider = ({ children }: Props) => {
     setBannerDeleting,
   } = useStore();
 
+  const fetchAllBanners = async () => {
+    setBannerLoading(true);
+    try {
+      const allBanners = await getBanners();
+      // sort by latest start date (assuming the most recent one should be active)
+      allBanners.sort(
+        (a: AdminBannerData, b: AdminBannerData) => b.startDate - a.startDate
+      );
```

Comment:
Since `allBanners` is already typed as `AdminBannerData[]`, TS already knows the types required to `sort` them, so the type annotations on the lambda aren't doing us any good.
```suggestion
      allBanners.sort((a, b) => b.startDate - a.startDate);
```

The same would apply to the `filter` and `sort` lambdas in `fetchAdminBanner()` below.